hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/AbstractSearchDataIntegrationTest.kt

```diff
@@ -0,0 +1,193 @@
+      .expectStatus().isOk
+      .expectBody(RestResponsePage::class.java)
+      .returnResult().responseBody
+
+    assertThat(response.numberOfElements).isEqualTo(expectedCount)
+    assertThat(response.content).size().isEqualTo(expectedPrisoners.size)
+    assertThat(response.content).extracting("prisonerNumber").containsAll(expectedPrisoners)
+  }
+
+  fun singlePrisonSearch(prisonSearch: PrisonSearch, fileAssert: String) {
+    webTestClient.post().uri("/prisoner-search/match")
+      .body(BodyInserters.fromValue(gson.toJson(prisonSearch)))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isOk
+      .expectBody().json(fileAssert.readResourceAsText())
+  }
+
+  fun prisonerMatch(matchRequest: MatchRequest, fileAssert: String) {
+    webTestClient.post().uri("/match-prisoners")
+      .body(BodyInserters.fromValue(gson.toJson(matchRequest)))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isOk
+      .expectBody().json(fileAssert.readResourceAsText())
+  }
+
+  fun possibleMatch(matchRequest: PossibleMatchCriteria, fileAssert: String) {
+    webTestClient.post().uri("/prisoner-search/possible-matches")
+      .body(BodyInserters.fromValue(gson.toJson(matchRequest)))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isOk
+      .expectBody().json(fileAssert.readResourceAsText(), true)
+  }
+
+  fun search(searchCriteria: SearchCriteria, fileAssert: String) {
+    search(searchCriteria).json(fileAssert.readResourceAsText())
+  }
+
+  fun search(searchCriteria: SearchCriteria): WebTestClient.BodyContentSpec =
+    webTestClient.post().uri("/prisoner-search/match-prisoners")
+      .body(BodyInserters.fromValue(gson.toJson(searchCriteria)))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isOk
+      .expectBody()
+
+  fun globalSearch(globalSearchCriteria: GlobalSearchCriteria, fileAssert: String) {
+    webTestClient.post().uri("/global-search")
+      .body(BodyInserters.fromValue(gson.toJson(globalSearchCriteria)))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isOk
+      .expectBody().json(fileAssert.readResourceAsText())
+  }
+  fun globalSearchPagination(globalSearchCriteria: GlobalSearchCriteria, size: Long, page: Long, fileAssert: String) {
+    webTestClient.post().uri("/global-search?size=$size&page=$page")
+      .body(BodyInserters.fromValue(gson.toJson(globalSearchCriteria)))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
```

Comment:
Some of these methods will be used in the next set of tests to be pulled through (They are just "lifted and shifted from prisoner-offender-search")